source/code/plugins/oms_configuration.rb

```diff
@@ -75,11 +81,84 @@ def parse_proxy_config(proxy_conf_str)
           Hash[ matches.names.map{ |name| name.to_sym}.zip( matches.captures ) ]
       end
 
+      def get_azure_resid_from_imds()
+          begin
+            uri = URI.parse(@@AzureIMDSEndpoint)
+            http_get_req = Net::HTTP::Get.new(uri, initheader = {'Metadata' => 'true'})
+
+            if @@ProxyConfig.empty?
+              http_req = Net::HTTP.new(uri.host, uri.port)
+            else
+              http_req = Net::HTTP.new(uri.host, uri.port, @@ProxyConfig[:addr], @@ProxyConfig[:port], @@ProxyConfig[:user], @@ProxyConfig[:pass])
+            end
+
+            http_req.open_timeout = 3
+            http_req.read_timeout = 2
+
+            res = http_req.start() do |http|
+              http.request(http_get_req)
+            end
+
+            imds_instance_json = JSON.parse(res.body)
+
+            return nil if !imds_instance_json.has_key?("compute") || imds_instance_json['compute'].empty? #classic vm
+
+            imds_instance_json_compute = imds_instance_json['compute']
+
+            #guard from missing keys
+            return nil unless imds_instance_json_compute.has_key?("subscriptionId") && imds_instance_json_compute.has_key?("resourceGroupName")
+                           && imds_instance_json_compute.has_key?("virtualMachines") && imds_instance_json_compute.has_key?("virtualMachineScaleSets")
```

Comment:
Shouldn't it be either VM or VMSS is present?

Reply:
sorry, got the keys names wrong. The response has only 'name' and 'vmScaleSetName' keys. The 'name' is name of VM if vmScaleSetName is empty and VMSS instance name otherwise.